### PR TITLE
add retention policy

### LINKED ISSUE #801: Add ability to specify a retention policy for storage buckets
Retention policy is supported by the underlying simple bucket module. See: https://github.com/terraform-google-modules/terraform-google-cloud-storage/tree/master/modules/simple_bucket

### PR DESCRIPTION
**Pull request by umairidris**
Fix #801

### CODE REVIEW

**Review by xingao267 — APPROVED** [review]

**Comment by xingao267** [comment]
cc @jordan83

File: docs/tfengine/schemas/resources.md [hunk 1]
```diff
@@ -1226,6 +1226,29 @@ Normalization will make all characters alphanumeric with underscores.
...
+Configuration of the bucket's data retention policy for how long
+objects in the bucket should be retained.
+
+Type: object
+
+### storage_buckets.retention_policy.is_locked
+
+If set to true, the bucket will be locked and permanently
+restrict edits to the bucket's retention policy.
+Caution: Locking a bucket is an irreversible action.
```
Comment (MartinPetkov) [inline docs/tfengine/schemas/resources.md:1240]: Is this true? A bucket becomes un-deletable once locked? Not even an owner or storage admin can unlock it?

**Review by MartinPetkov — APPROVED** [review]

File: docs/tfengine/schemas/resources.md (hunk 1 above)
Comment (umairidris) [reply docs/tfengine/schemas/resources.md:1240]: I think so from what I played around with, but maybe you can get away with just deleting the project.

Comment (xingao267) [reply docs/tfengine/schemas/resources.md:1240]: Maybe add the link https://cloud.google.com/storage/docs/bucket-lock#overview here as well?

File: templates/tfengine/components/resources/storage_buckets/main.tf [hunk 2]
```diff
@@ -49,6 +49,13 @@ module "{{resourceName . "name"}}" {
   ]
   {{end -}}
 
+  {{if has . "retention_policy" -}}
+  retention_policy = {
+    is_locked        = {{.retention_policy.is_locked}}
```
Comment (xingao267) [inline templates/tfengine/components/resources/storage_buckets/main.tf:54]: can we make is_locked optional? and default to false?

**Comment by umairidris** [comment]
@xingao267 good to merge?

**Comment by xingao267** [comment]
@umairidris yep, i'll merge it

**Pull request merged**